Neo4jClient/Serialization/CommonDeserializerMethods.cs

```diff
@@ -19,10 +19,22 @@ class CommonDeserializerMethods
         public static string RemoveResultsFromJson(string content)
         {
             var root = JToken.Parse(content);
+            var errors = root.Value<JToken>("errors");
+            if (errors != null && errors.HasValues)
+            {
+                throw DeserializeNeo4jError(errors);
+            }
             var output = root.SelectTokens("$.results[0]").Select(j => j.ToString(Formatting.None)).FirstOrDefault();
             return output;
         }
 
+        public static NeoException DeserializeNeo4jError(JToken error) =>
+            new NeoException(new ApiModels.ExceptionResponse
+            {
+                Exception = error.First.Value<JToken>("code").ToString(),

```

Comment:
Can the value of `error.First.Value<JToken>("code").` be null? Does this need to be:

`error.First.Value<JToken>("code")?.ToString()`

What about `error.First`?

Reply:
Also - is this the same error structure for `3.x` and `4.x` servers?

Reply:
Thanks for the response, but busy atm, will check this when I get the chance.

Reply:
The error structure appears to be the same for 3.x and 4.x: https://neo4j.com/docs/http-api/3.5/actions/handling-errors/ 

I have added improved null handling with some test cases for when "message" or "code" is null. 

I think `errors != null && errors.HasValues` protects us from `error.First` being null, and either way, I think it is best not to throw a neoexception in that case as we don't have any info about it.

Reply:
Another point is that if any exception is thrown here it is still handled in the old way with the "Neo4jClient was unable to deserialize..." message.

Reply:
@cskardon let me know what you think.

Reply:
Hey Angus, sorry for the time! Looks good to me! Thank you